### PR TITLE
Teams in add team Popup are not filtered by teams with participation

### LINKED ISSUE #300: Teams in add team Popup are not filtered by teams with participation
It can lead to error `Unique constraint failed on the fields: (`userId`,`groupId`)`
image

### PR DESCRIPTION
**Pull request by Katrin-kudryash**
## PR includes

-   [x] Bug Fix

## Related issues

Resolve #300

### CODE REVIEW

**Comment by Katrin-kudryash** [comment]
https://github.com/taskany-inc/crew/assets/81121562/402d8c3a-ccd6-4e6d-bb4f-e57750fbe3c1

File: src/components/InlineGroupSelectForm.tsx [hunk 1]
```diff
@@ -19,12 +19,13 @@ type InlineGroupSelectFormProps = {
...
     actionText: string;
     onSubmit: (group: Group) => Promise<void>;
+    filter?: string[];
 };
 
 export const InlineGroupSelectForm = (props: InlineGroupSelectFormProps) => {
     const [search, setSearch] = useState('');
     const [selectedGroup, setSelectedGroup] = useState<Group>();
-    const groupListQuery = trpc.group.getList.useQuery({ search, take: 10 });
+    const groupListQuery = trpc.group.getList.useQuery({ search, take: 10, filter: props.filter || [] });
```
Comment (tsumo) [inline src/components/InlineGroupSelectForm.tsx:28]: dont need `|| []` here

File: src/components/UserMembershipsList/UserMembershipsList.tsx [hunk 2]
```diff
@@ -28,6 +29,10 @@ export const UserMembershipsList = ({ user }: UserMembershipsProps) => {
     const onAddUserToTeam = async (group: Group) => {
         await addUserToGroup.mutateAsync({ userId: user.id, groupId: group.id });
     };
+    const groupFilter = useMemo(() => {
+        const membershipIds = user.memberships.map(({ groupId }) => groupId);
```
Comment (tsumo) [inline src/components/UserMembershipsList/UserMembershipsList.tsx:33]: dont need const, just return

File: src/modules/groupMethods.ts [hunk 3]
```diff
@@ -105,10 +105,16 @@ export const groupMethods = {
     },
 
     getList: (data: GetGroupList) => {
-        return prisma.group.findMany({
-            where: { name: { contains: data.search, mode: 'insensitive' }, archived: false },
+        const groupList = prisma.group.findMany({
```
Comment (tsumo) [inline src/modules/groupMethods.ts:108]: dont need const, just return

**Review by tsumo — APPROVED** [review]

**Review by nyet-ty — APPROVED** [review]

**Pull request merged**